### PR TITLE
Move dataset meta tab to another bean.

### PR DESCRIPTION
**Pull request by MrK191**
- Plus removed some unused methods.

### CODE REVIEW

File: dataverse-webapp/src/main/webapp/metadataTab.xhtml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            </ul>
+        </div>
+    </div>
+    <ui:include src="viewMetadata.xhtml">
+        <ui:param name="metadataBlocks"
+                  value="#{DatasetMetadataTab.metadataBlocks.entrySet().toArray()}"/>
+        <ui:param name="globalId" value="#{DatasetMetadataTab.dataset.globalIdString}"/>
+        <ui:param name="altPID" value="#{DatasetMetadataTab.dataset.alternativePersistentIdentifier}"/>
+        <ui:param name="publicationDate"
+                  value="#{DatasetMetadataTab.dataset.publicationDate != null ? DatasetMetadataTab.dataset.publicationDateFormattedYYYYMMDD : ''}"/>
```
Comment (madryk) [inline dataverse-webapp/src/main/webapp/metadataTab.xhtml]: Most probably we can skip checking of publicationDate and go straight for DatasetMetadataTab.dataset.publicationDateFormattedYYYYMMDD

File: dataverse-webapp/src/main/webapp/metadataTab.xhtml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                xmlns:ui="http://java.sun.com/jsf/facelets"
+                xmlns:p="http://primefaces.org/ui"
+                xmlns:jsf="http://xmlns.jcp.org/jsf"
+>
+
+    <div class="button-block tab-header margin-bottom text-right">
+        <p:link styleClass="btn btn-default btn-access" disabled="#{DatasetMetadataTab.datasetLocked}"
+                outcome="editDatasetMetadata"
+                rendered="#{!widgetWrapper.widgetView and dataverseSession.getUser().isAuthenticated()
+                                                               and DatasetMetadataTab.canUpdateDataset() and !DatasetMetadataTab.dataset.deaccessioned}">
```
Comment (madryk) [inline dataverse-webapp/src/main/webapp/metadataTab.xhtml]: There is no need to check both isAuthenticated and canUpdate

File: dataverse-webapp/src/main/webapp/metadataTab.xhtml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+<ui:composition xmlns="http://www.w3.org/1999/xhtml"
+                xmlns:h="http://java.sun.com/jsf/html"
+                xmlns:f="http://java.sun.com/jsf/core"
+                xmlns:ui="http://java.sun.com/jsf/facelets"
+                xmlns:p="http://primefaces.org/ui"
+                xmlns:jsf="http://xmlns.jcp.org/jsf"
+>
+
+    <div class="button-block tab-header margin-bottom text-right">
+        <p:link styleClass="btn btn-default btn-access" disabled="#{DatasetMetadataTab.datasetLocked}"
```
Comment (madryk) [inline dataverse-webapp/src/main/webapp/metadataTab.xhtml]: I think that lock state can be changed during the lifetime of the view. On dataset page we have some refreshing code. When initially dataset is locked then in 1.5s intervals there is a check if lock is still present. However in this tab there always will be initial lock state.

**Review by madryk — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/tab/DatasetMetadataTab.java [hunk 4]
```diff
@@ -80,6 +81,15 @@ public boolean canUpdateDataset() {
         return permissionsWrapper.canUpdateDataset(dvRequestService.getDataverseRequest(), this.dataset);
     }
 
+    /**
+     * Updates the dataset lock state and refreshes button in metadataTab xhtml.
+     */
+    public boolean updateDatasetLockState(boolean isDatasetLocked) {
+        this.isDatasetLocked = isDatasetLocked;
+        PrimeFaces.current().ajax().update("datasetForm:tabView:editMetadataButton");
```
Comment (madryk) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/tab/DatasetMetadataTab.java]: Not sure if update here is elegant. We depend here very tightly on component names in xhtml. It is easy to miss this when for example we want to change id's of some component. It is defined in xhtml and some handling is in java classes.
I am wondering if we could change update attribute of `<p:commandButton id="refreshButton" process="@this" widgetVar="refreshButton" update="...` on `dataset.xhtml`.

**Review by madryk — CHANGES_REQUESTED** [review]

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/tab/DatasetMetadataTab.java (hunk 4 above)
Comment (MrK191) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/tab/DatasetMetadataTab.java]: I've made a temporary button to see if it's working, so it should be working.

**Review by madryk — APPROVED** [review]

**Pull request merged**